### ISSUE TITLE
Why does null.Time text-marshaling return null string?

### ISSUE THREAD

**Issue by limoli** [guregu/null#47]
Testing my models on GraphQL, I detected some issues regarding **text-marshaling** implementation in this library.

**null.Time returns "null"**
```
func (t Time) MarshalText() ([]byte, error) {
	if !t.Valid {
		return []byte("null"), nil
	}
	return t.Time.MarshalText()
}
```

**Other null.X implementations**
```
// MarshalText implements encoding.TextMarshaler.
// It will encode a blank string when this String is null.
func (s String) MarshalText() ([]byte, error) {
	if !s.Valid {
		return []byte{}, nil
	}
	return []byte(s.String), nil
}

// MarshalText implements encoding.TextMarshaler.
// It will encode a blank string if this Int is null.
func (i Int) MarshalText() ([]byte, error) {
	if !i.Valid {
		return []byte{}, nil
	}
	return []byte(strconv.FormatInt(i.Int64, 10)), nil
}
```

My question/curiosity is: why does null.Time text-marshaling return a "null" string instead of an empty string like the others?

**Comment by guregu** [comment]
Good question...
I don't have a real answer. It's because I accepted a PR from someone without checking it closely enough.
I should probably release v4 and fix this.

**Comment by limoli** [comment]
Thank you so much @guregu ! :)

**Comment by guregu** [comment]
This is fixed in `v4.0.0`, just released.
